### DISCUSSION TITLE
Static variables shared between model, simulation and experiment

### DISCUSSION THREAD

**Discussion by chapuisk** [gama-platform/gama#63] [Ideas]
When a model start to be of a reasonable size, I often use a gaml file to store all kind of constant variables. Those includes initial values, all possible categorical values, "super" parameter (e.g. vector of booleans turn into categorial instance)... well all sort of things that I use to initialize the simulation.

I just notice that when simulations are not kept in memory (keep_simulation false in experiment - see issue #2727 ) those values are not accessible anymore from the experiment. This might not be a problem in most cases. However it is particularly problematic when automatic exploration methods are used, e.g. sampling methods with among facet on parameters or path to output file define in the model. It requires to redefine all non numerical variables, like categorical / nominal or complex variables, or to duplicate them (no one want to duplicate code).

Then I wander if it would be a good idea to have some kind of special variable that would be close to the concept of static variable in Java or other POO languages.

Just leave this idea here to not forget ;)

**Comment by lesquoyb** [comment]
Has your issue regarding the values not being accessible in experiment when keep_simulation is false been solved ? (the issue you point out is marked as solved but not sure it fix your peculiar problem)
